### PR TITLE
Parallelize operations in groups of pods

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Paralelize updates to pod objects. These operations can be slow, as they require a connection with the apiserver.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/util/parallelize/parallelize.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+package parallelize
+
+import (
+	"context"
+
+	"k8s.io/client-go/util/workqueue"
+)
+
+const maxParallelism = 8
```
Comment (alculquicondor) [inline pkg/util/parallelize/parallelize.go:25]: Using a conservative number here. In the future, this could be configurable.

**Comment by alculquicondor** [comment]
E2E tests passed 2/5 times

**Review by mimowo — COMMENTED** [review]
/lgtm

**Pull request merged**